### PR TITLE
[RECONSTRUCTION] [CLANG] Fixes warnings reported by clang 14

### PR DESCRIPTION
**Pull request by smuzaffar**
This should fix the clang 14 warning. For now I have removed the unused `res` variable but if needed then I can just print out the `res` value so that  compiler do not complain about unused variable.

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by mandrenguyen** [comment]
+reconstruction
Technical, no changes to reco

File: CommonTools/Utils/test/testExpressionParser.cc [hunk 1]
```diff
@@ -264,10 +264,9 @@ void testExpressionParser::checkAll() {
...
       expr.reset();
       reco::parser::expressionParser<pat::Muon>("triggerObjectMatchesByPath('HLT_Something').size()", expr, b);
-      double res = 0;
       for (size_t i = 0; i < 10 * 1000; ++i) {
         for (size_t j = 0; j < 100; ++j) {
-          res += expr->value(o);
+          expr->value(o);
           break;
         }
         break;
```
Comment (perrotta) [inline CommonTools/Utils/test/testExpressionParser.cc:272]: Given this `break`, what is the purpose of the `outer` for loop?

Comment (smuzaffar) [reply CommonTools/Utils/test/testExpressionParser.cc:272]: I was also confused by this may be @VinInn  can help here

**Comment by perrotta** [comment]
@VinInn could you please comment about https://github.com/cms-sw/cmssw/pull/40516#pullrequestreview-1249316883, so that we can conclude with this technical PR?

**Comment by rappoccio** [comment]
@VinInn ping about this review comment?

**Comment by VinInn** [comment]
Sorry: what I have to do with this? Not my code. and do not understand the two ```break``` either

**Comment by smuzaffar** [comment]
sorry @VinInn my bad , I mixed up `ExpressionEvaluator` (which is mostly developeed by you) and `ExpressionParser`.

**Comment by perrotta** [comment]
@smuzaffar so what should we do here?
Your PR fixes the clang warning, as intended. As such, I'd have no objections in merging it.
On the other hand, it pointegs to a test which apparently nobody can understand: is it testing anything, then?
If anyone knows or understands how was this test supposed to work, I'd preferably fix it. Otherwise, we can just merge it and forget afterwards.

**Comment by perrotta** [comment]
> @perrotta , the extra `break` statements were added [many years ago](https://cmscvs.web.cern.ch/cgi/viewvc.cgi/cvsroot/CMSSW/CommonTools/Utils/test/testExpressionParser.cc?r1=1.26&r2=1.27) . As no one has complained about these extra `break` statements so I would suggest that we also just clean up the outer loop. If agree then I can update this PR

Thank you @smuzaffar: yes, that's perfectly fine with me.
At the end, the inner and outer loop may coalesce into the simple single line
```C++
      expr->value(o);
```

**Comment by smuzaffar** [comment]
please test

**Comment by mandrenguyen** [comment]
+reconstruction

**Comment by perrotta** [comment]
+1

**Pull request merged**